src/models/subscriptiondata.cpp

```diff
@@ -0,0 +1,189 @@
+  m_planCurrency = planData["currency"].toString().toUpper();
+  if (m_planCurrency.isEmpty()) {
+    return false;
+  }
+  logger.debug() << "Parse plan ready";
+
+  // Payment
+  QJsonObject paymentData = obj["payment"].toObject();
+
+  // If we do not receive payment data from FxA we don’t show it instead of
+  // throwing an error. There is a known bug with FxA which causes FxA to not
+  // show payment info: https://mozilla-hub.atlassian.net/browse/FXA-3856.
+  if (!paymentData.isEmpty()) {
+    // Payment provider
+    m_paymentProvider = paymentData["payment_provider"].toString();
+    // We should always get a payment provider if there is payment data
+    if (m_paymentProvider.isEmpty()) {
+      return false;
+    }
+
+    // Payment type
+    m_paymentType = paymentData["payment_type"].toString();
+
+    // For credit cards we also show card details
+    if (m_paymentType == "credit") {
+      m_creditCardBrand = paymentData["brand"].toString();
+      if (m_creditCardBrand.isEmpty()) {
+        return false;
+      }
+
+      m_creditCardLast4 = paymentData["last4"].toString();
+      if (m_creditCardLast4.isEmpty()) {
+        return false;
+      }
+
+      m_creditCardExpMonth = paymentData["exp_month"].toInt();
+      if (!m_creditCardExpMonth) {
+        return false;
+      }
+
+      m_creditCardExpYear = paymentData["exp_year"].toInt();
+      if (!m_creditCardExpYear) {
+        return false;
+      }
+    }
+  }
+
+  // Subscription
+  QJsonObject subscriptionData = obj["subscription"].toObject();
+
+  QString type = subscriptionData["_subscription_type"].toString();
+  if (type.isEmpty()) {
+    return false;
+  }
+
+  // Enum from string
+  bool ok = false;
+  m_type = static_cast<TypeSubscription>(
+      QMetaEnum::fromType<TypeSubscription>().keyToValue(type.toUtf8(), &ok));
+  if (!ok) {
+    logger.error() << "Unsupported subscription type:" << type;
+    return false;
+  }
+
+  // Parse subscription data depending on subscription platform
+  switch (m_type) {
+    case web:
+      if (!parseSubscriptionDataWeb(subscriptionData)) {
+        return false;
+      }
+      break;
+    case iap_apple:
+      [[fallthrough]];
+    case iap_google:
+      if (!parseSubscriptionDataIap(subscriptionData)) {
+        return false;
+      }
+      break;
+    default:
+      logger.error() << "No matching subscription type" << type;
+      return false;
+  }
+
+  m_rawJson = json;
+  emit changed();
+  logger.debug() << "Subscription data from JSON ready";
+
+  return true;
+}
+
+bool SubscriptionData::parseSubscriptionDataIap(
+    const QJsonObject& subscriptionData) {
+  logger.debug() << "Parse IAP start" << m_type;
+
+  m_expiresOn = subscriptionData["expiry_time_millis"].toVariant().toLongLong();
+  if (!m_expiresOn) {
+    return false;
+  }
+  m_isCancelled = !subscriptionData["auto_renewing"].toBool();
+
+  quint64 now = QDateTime::currentMSecsSinceEpoch();
+  if (now <= m_expiresOn) {
+    m_status = "active";
```

Comment:
I would like to see an enum here. m_status should not be parsed twice (here and in the QML code). Follow-up.